CommonCompiler.cmake

```diff
@@ -1,12 +1,13 @@
-# Copyright (c) 2012-2014 Fabien Delalondre <fabien.delalondre@epfl.ch>
+# Copyright (c) 2012-2016 Fabien Delalondre <fabien.delalondre@epfl.ch>
 #                         Stefan.Eilemann@epfl.ch
 #
 # Sets compiler optimization, definition and warnings according to
 # chosen compiler. Supported compilers are XL, Intel, Clang, gcc (4.4
 # or later) and Visual Studio (2008 or later).
 #
-# This defines the common_compile_options() function to apply compiler flags and
-# features for the given target.
+# This defines the common_compile_options() function to apply compiler
+# flags and features for the given target, and adds the target to the
+# project-all meta target.
```

Comment:
I don't think adding the target to the project-all target has anything to do with the compile options (setting compilation flags).
I would move this to the three places where this function is needed: CommonApplication, CommonLibrary, and CommonCPPCtest, probably where  ${PROJECT_NAME}_ALL_DEP_TARGETS is already filled.